install.bat

```diff
@@ -507,8 +540,18 @@ local function get_msvc_env_setup_cmd()
 	assert(type(vars.UNAME_M) == "string", "requires vars.UNAME_M to be set before calling this function.")
 	local x64 = vars.UNAME_M=="x86_64"
 
-	-- 1. try visual studio command line tools
-	local vcdir = get_visual_studio_directory()
+	-- 1. try visual studio command line tools of VS 2017 or higher
+	local vsdir = get_visual_studio_directory_from_vswhere()
+	if vsdir then
+		local vcvarsall = vsdir .. '\\VC\\Auxiliary\\Build\\vcvarsall.bat'
+		if exists(vcvarsall) then
+			local vcvarsall_args = { x86 = "", x86_64 = " x64" }
+			return ('call "%s"%s'):format(vcvarsall, vcvarsall_args[vars.UNAME_M])

```

Comment:
This line appears to have a windows line ending on it.

Reply:
I double-checked, the whole file has CRLF line ending. I guess this is by design because it is is bat file running on WIndows. Github shows `^M` on this line because it parses it wrong. You can see `^M` on line 536, 545, 719, 958 of master branch `install.bat` as well.

Reply:
Weird.
Should report a bug to github :smile:

Reply:
Because it parses the file as bat....